cmd/proxima/proxima.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"flag"
+	"github.com/Symantec/Dominator/lib/fsutil"
+	"github.com/Symantec/scotty/lib/apiutil"
+	"github.com/influxdata/influxdb/models"
+	"github.com/influxdata/influxdb/uuid"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+)
+
+func setHeader(w http.ResponseWriter, r *http.Request, key, value string) {
+	r.Header.Set(key, value)
+	w.Header().Set(key, value)
+}
+
+func uuidHandler(inner http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		uid := uuid.TimeUUID()
+		setHeader(w, r, "Request-Id", uid.String())
+		setHeader(w, r, "X-Influxdb-Version", "0.13.0")
+		inner.ServeHTTP(w, r)
+	})
+}
+
+type seriesListType struct {
+	Series []models.Row `json:"series"`
+}
+
+type resultListType struct {
+	Results []seriesListType `json:"results"`
+}
+
+func main() {
+	fConfigFile := flag.String("config", "./proxima.config", "config file")
+	fPort := flag.String("port", ":8086", "listen port")
+	flag.Parse()
+	executer := newExecuter()
+	logger := log.New(os.Stderr, "", log.LstdFlags)
+	changeCh := fsutil.WatchFile(*fConfigFile, logger)
+	// We want to be sure we have something valid in the config file
+	// initially.
+	select {
+	case readCloser := <-changeCh:
+		if err := executer.SetupWithStream(readCloser); err != nil {
+			logger.Fatal(err)
+		}
+		readCloser.Close()
+	case <-time.After(time.Second):
+		logger.Fatal("No config file")
+	}
+	go func() {
+		for readCloser := range changeCh {
+			if err := executer.SetupWithStream(readCloser); err != nil {
+				logger.Println(err)
+			}
+			readCloser.Close()
+		}
+	}()
+	http.Handle(
+		"/query",
+		uuidHandler(
```

Comment:
Made this more like what I do in scotty. Kept the closure, but closure short because I delegate to another named  function. I chose to keep the closure because its silly to have a named function take a url.Values and return an (interface{}, error) as that is an artefact of http. 

So my named function takes an executer, query, epoch, and database name and returns an actual response instead of an interface{}.